NitroxClient/Communication/Packets/Processors/DiscordRequestIPProcessor.cs

```diff
@@ -1,13 +1,33 @@
-﻿using NitroxClient.Communication.Packets.Processors.Abstract;
+﻿using System.Net;
+using System.Threading.Tasks;
+using NitroxClient.Communication.Packets.Processors.Abstract;
 using NitroxClient.MonoBehaviours.Discord;
+using NitroxModel.Helper;
 using NitroxModel.Packets;
 
 namespace NitroxClient.Communication.Packets.Processors;
 
 public class DiscordRequestIPProcessor : ClientPacketProcessor<DiscordRequestIP>
 {
-    public override void Process(DiscordRequestIP packet)
+    public override async void Process(DiscordRequestIP packet)
     {
-        DiscordClient.UpdateIpPort(packet.IpPort);
+        string ipPort = await GetIpPortAsync();
+        DiscordClient.UpdateIpPort(ipPort);
+    }
+
+    /** Get the WAN IP address or the Hamachi IP address if the WAN IP address is not available. */
+    private static async Task<string> GetIpPortAsync()
+    {
+        Task<IPAddress> wanIp = NetHelper.GetWanIpAsync();
+        Task<IPAddress> hamachiIp = Task.Run(NetHelper.GetHamachiIp);
+        if (await wanIp != null)
+        {
+            return wanIp.Result.ToString();
+        }
+        if (await hamachiIp != null)
+        {
+            return hamachiIp.Result.ToString();
+        }
+        return "127.0.0.1";
```

Comment:
Don't think we should return the local IP here. Two different discord client on the same machine sound strange.

Reply:
Yep. It makes sense if on client-side, but this code should be running server-side. So returning localhost is no good.